rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,327 @@
+-- Your SQL goes here
+
+
+-- Create the schema
+CREATE SCHEMA IF NOT EXISTS legacy_migration_v1;
+
+-- Replace `move_resources` with account transactions
+CREATE OR REPLACE VIEW legacy_migration_v1.move_resources AS
+    SELECT *
+    FROM account_transactions at2
+  ;
+-- CREATE INDEX lm1_mr_ver_index ON legacy_migration_v1.move_resources USING btree (transaction_version DESC)
+
+
+-- replace `coin_activities` with `fungible_asset_activities`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_activities AS
+    SElECT
+        transaction_version,
+        owner_address as event_account_address,
+        -- these two below are mildly concerning
+        0 as event_creation_number,
+        0 as event_sequence_number,
+        event_index,
+        owner_address,
+        storage_id,
+        asset_type as coin_type,
+        is_frozen,
+        amount,
+        "type" as activity_type,
+        is_gas_fee,
+        gas_fee_payer_address,
+        is_transaction_success,
+        entry_function_id_str,
+        block_height,
+        token_standard,
+        transaction_timestamp,
+        inserted_at,
+        storage_refund_amount
+    FROM public.fungible_asset_activities
+    WHERE token_standard = 'v1'
+  ;
+-- CREATE INDEX lm1_ca_ct_a_index ON legacy_migration_v1.coin_activities USING btree (coin_type, amount)
+-- CREATE INDEX lm1_ca_ct_at_a_index ON legacy_migration_v1.coin_activities USING btree (coin_type, activity_type, amount)
+-- CREATE INDEX lm1_ca_oa_ct_at_index ON legacy_migration_v1.coin_activities USING btree (owner_address, coin_type, activity_type, amount)
+-- CREATE INDEX lm1_ca_oa_igf_index ON legacy_migration_v1.coin_activities USING btree (owner_address, is_gas_fee)
+
+-- replace `coin_balances` with `fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_balances AS
+    SELECT
+        transaction_version,
+        owner_address,
+        -- this is mainly for hashing the coin type for primary key
+        substring(asset_type, 66) as coin_type_hash,
```

Comment:
why are we truncating this? you can have `0x1::aptos_coin::AptosCoin` as asset_type
we should note somewhere that coin_balances.coin_type_hash is going to change

Reply:
actually this will do the mapping
```sql
encode(sha256(asset_type::bytea), 'hex')
```

Reply:
what do you mean it's going to change? Is there a way to leave it as is?

Reply:
i think we stopped using _hash fields in v2 since we have proper _id fields, for the purpose of backwards compatibility, not sure if we should put the _id fields in (since purpose of column should be unique identifier) which would change the value or hash to keep value the same

Reply:
nvm i see, the _id fields are just 0x + _hash fields